forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt

```diff
@@ -67,22 +83,45 @@ class EncryptedPreferences {
         }
 
         // Creates the instance for the encrypted preferences.
-        private fun createPreferencesFile(context: Context,
-                                              fileName: String,
-                                              aliasName: String): SharedPreferences {
-
-
-            val masterKeyAlias = MasterKey.Builder(context, aliasName)
-                .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-                .build()
-
-            return EncryptedSharedPreferences.create(
+        private fun createPreferencesFile(
+            context: Context,
+            fileName: String,
+            aliasName: String? = null
+        ): SharedPreferences {
+            val builder =
+                aliasName?.let { MasterKey.Builder(context, it) } ?: MasterKey.Builder(context)
+            builder.setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+            return  EncryptedSharedPreferences.create(
                 context,
                 fileName,
-                masterKeyAlias,
+                builder.build(),
                 EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                 EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
             )
         }
+
+        // master key recovery code
+        private fun recoverFromDefaultMasterKey(
+            context: Context,
+            fileName: String = "secret_shared_prefs" + context.packageName
+        ): MutableMap<String, MutableSet<String>> {
+            val cache = mutableMapOf<String, MutableSet<String>>()
+            // creating all 1.0.5 alpha files
+            try {
+                val secureSharedPreferences =
+                    createPreferencesFile(context, fileName)
+                secureSharedPreferences.all.entries.map { it.key }.forEach { key ->
+                    secureSharedPreferences.getStringSet(key, emptySet())?.let {
```

Comment:
Just to confirm, we previously set data as StringSet?

Reply:
yes , this is how we set it before in the WebAuthnDataRepository

```       encryptedPreferences.edit().apply {
                map.forEach {
                    this.putStringSet(it.key, it.value)
                }
            }.apply()
```